src/components/HomePage.js

```diff
@@ -28,23 +31,29 @@ function ServerProps({serverInfo}) {
   );
 }
 
-function SessionInfo(props) {
-  const {session} = props;
-  const {busy, serverInfo} = session;
+function SessionInfo({session: {busy, serverInfo}}) {
   return (
     <div>
       {busy ? <Spinner /> :
         <div className="panel server-info-panel panel-default">
           <div className="panel-heading"><b>Server information</b></div>
           <div className="panel-body">
-            <ServerProps serverInfo={serverInfo} />
+            <ServerProps node={serverInfo} />
           </div>
         </div>}
     </div>
   );
 }
 
 export default class HomePage extends Component {
+  props: {
+    session: SessionState,
+    history: string[],
+    clearHistory: Function,
+    setup: Function,
+    navigateToExternalAuth: Function,
```

Comment:
General remark with function types, would be useful to actually type their signatures :D